python/cudf/cudf/core/index.py

```diff
@@ -2343,6 +2344,24 @@ def day_of_year(self):
         """
         return self._get_dt_field("day_of_year")
 
+    @property
+    def is_leap_year(self):
+        """
+        Boolean indicator if the date belongs to a leap year.
+
+        A leap year is a year, which has 366 days (instead of 365) including
+        29th of February as an intercalary day. Leap years are years which are
+        multiples of four with the exception of years divisible by 100 but not
+        by 400.
+
+        Returns
+        -------
+        ndarray
+        Booleans indicating if dates belong to a leap year.
+        """
+        res = is_leap_year(self._values).fillna(False)
+        return cupy.array(res.to_gpu_array())
```

Comment:
This makes a copy by default -- use `cp.asarray()` if we don't want that.

Reply:
Do we need to go cudf -> numba -> cupy? I think it should be faster if we go cudf -> cupy directly with either `cp.asarray(res)` relying on the cuda array interface or with `cp.asarray(res.values)`.

Reply:
Agree -- if it's possible (and I don't see why not), we should call `asarray()` directly on `res`